cylc/flow/scheduler.py

```diff
@@ -1745,19 +1746,15 @@ async def _main_loop(self) -> None:
         if self.xtrigger_mgr.do_housekeeping:
             self.xtrigger_mgr.housekeep(self.pool.get_tasks())
 
-        self.pool.clock_expire_tasks()
-        self.release_queued_tasks()
-
-        if (
-            self.get_run_mode() == RunMode.SIMULATION
-            and sim_time_check(
-                self.task_events_mgr,
-                self.pool.get_tasks(),
-                self.workflow_db_mgr,
-            )
+        if sim_time_check(
+            self.task_events_mgr,
+            self.pool.get_tasks(),
+            self.workflow_db_mgr,
         ):
             # A simulated task state change occurred.
             self.reset_inactivity_timer()
+        self.pool.clock_expire_tasks()
+        self.release_queued_tasks()
```

Comment:
Why this change in order?

Reply:
No idea. I don't think that it was intentional.